scripts/serve_header/README.md

```diff
@@ -0,0 +1,81 @@
+serve_header.py
+===============
+
+Serves the `single_include/nlohmann/json.hpp` header file over HTTP(S).
+
+The header file is automatically amalgamated on demand.
+
+## Prerequisites
+
+> **Note:** All file locations are relative to the project root folder and all commands are assumed to be executed from the same root folder.
+
+1. Make sure these Python packages are installed.
+    ```
+    PyYAML
```

Comment:
Please create a `requirements.txt`.

Reply:
Should I then also setup venv? I thought that was a bit much for one package (who doesn't have PyYAML installed already?)

Reply:
I never install any packages outside a venv. And I would never expect a script to require any package unless I see a requirements.txt file.

Maybe it's a matter of style, but dropping a requirements.txt would certainly make it clearer what's needed.

Reply:
Well, I am a Gentoo user and our package manager is written in Python, so that might explain why I expect a lot of Python packages to be installed system-wide. :-)

I was just copying from the `mkdocs/requirements.txt` and noticed that `watchdog==2.1.7` is also already in there (of course `PyYAML` is there because of `mkdocs.yml`).

Would it then maybe make sense to hoist the `docs/mkdocs/venv` folder into the root and share it? That was my original plan but I ultimately decided against it.

Reply:
If the requirements ever change or the versions don't match that might create problems.

Reply:
No, mixing is not a good idea. The documentation is already such a wonderland of requirements (one with a fixed version due to a bug), that I would not like to mix it with other things.